numpy/f2py/cfuncs.py

```diff
@@ -870,6 +870,19 @@
 }
 """
 
+needs['int64_t_from_pyobj'] = ['long_from_pyobj']
+cfuncs['int64_t_from_pyobj'] = """
+static int
+int64_t_from_pyobj(int64_t* v, PyObject *obj, const char *errmess) {
+    long i = 0;
+    if (long_from_pyobj(&i, obj, errmess)) {
+        *v = (int64_t)i;
+        return 1;
+    }
+    return 0;
+}
+"""
```

Comment:
@Pranavchiku could you check (in light of the previous comment) which of these need to be explicitly written out and which can be just type cast? I suppose it might be best to just have each one handled explicitly, if only to get the C++ types correct in the function signatures....

Reply:
Yes, on it.

Reply:
Done, we already have ones for `float/double_complex` so I skipped both. I am not sure about how to add it for bool hence skipped it.